### DISCUSSION TITLE
Virtualizable support

### DISCUSSION THREAD

**Discussion by bartoszgolebiowski** [TanStack/table#2153] [General] [answered]
Is there any Virtualizable support, I am trying to find something in docs, but I can not see anything.
https://react-table.js.org/

**Comment by atardito** [comment, marked as answer]
There is an example:
[https://react-table.js.org/examples/ui](https://react-table.js.org/examples/ui)

**Reply by atardito** [reply]
Another example but with Sub Component:
[https://codesandbox.io/s/sy9tb?file=/src/App.js](https://codesandbox.io/s/sy9tb?file=/src/App.js)

**Comment by omridevk** [comment]
I think it is important to make sure not to use any layout that generates TH/TR/table etc, and only divs. like useFlexLayout or useAbsoluteLayout.

**Reply by cahehe** [reply]
How come th/tr/table are not used?

**Comment by bartoszgolebiowski** [comment]
Thank you all for comments.
I have last question. Do I need to install other dependency than react table to support virtualization?

**Reply by omridevk** [reply]
The example uses https://github.com/bvaughn/react-window

**Reply by omridevk** [reply]
which is a more light weight version of https://github.com/bvaughn/react-virtualized
but you can use both.

**Reply by omridevk** [reply]
you need to install it, and make the adjusted code changes needed, i.e not render all the rows, but create a callback that will tell react-window how to render every row when it comes into view.
see line 64 in the demo
https://react-table.js.org/examples/ui

**Reply by omridevk** [reply]
and react-window usually must go together with https://www.npmjs.com/package/react-virtualized-auto-sizer
otherwise your table size won't be adjusted when browser is resized.
